### ISSUE TITLE
dubiously overconstrained warnings about project-level configuration file name

### ISSUE THREAD

**Issue by jclulow** [rust-lang/cargo#13912]
### Problem

I updated to **cargo 1.78.0 (54d8815d0 2024-03-26)** today and attempted a **cargo clean** in an older project.  I was greeted with this warning:

```
warning: `/ws/facade/.cargo/config` is deprecated in favor of `config.toml`
note: if you need to support cargo 1.38 or earlier, you can symlink `config` to `config.toml`
```

Despite the fact that there's really no reason beyond potentially aesthetics not to just continue to read `.cargo/config` if it's there, I would like the warning to go away, so I did what it told me to do:

```
 $ ls -l .cargo/
total 2
lrwxrwxrwx   1 jclulow  staff         11 May 13 16:03 config -> config.toml
-rw-r--r--   1 jclulow  staff         41 Dec  6  2022 config.toml
```

Now I get a _new_ warning, which seems, at best, unsporting:

```
warning: both `/ws/facade/.cargo/config` and `/ws/facade/.cargo/config.toml` exist. Using `/ws/facade/.cargo/config`
```

### Possible Solution(s)

I suspect that perhaps the second warning could at least not misfire in the case that one has done as directed by `cargo` with respect to symlinks.

### Version

```text
$ cargo version --verbose
warning: both `/ws/facade/.cargo/config` and `/ws/facade/.cargo/config.toml` exist. Using `/ws/facade/.cargo/config`
cargo 1.78.0 (54d8815d0 2024-03-26)
release: 1.78.0
commit-hash: 54d8815d04fa3816edc207bbc4dd36bf18014dbc
commit-date: 2024-03-26
host: x86_64-unknown-illumos
libgit2: 1.7.2 (sys:0.18.2 vendored)
libcurl: 8.6.0-DEV (sys:0.4.72+curl-8.6.0 vendored ssl:OpenSSL/1.1.1w)
ssl: OpenSSL 1.1.1w  11 Sep 2023
os: illumos helios-2.0.22365 [64-bit]
```

**Comment by weihanglo** [comment]
Thanks for the report. This should already be resolved by https://github.com/rust-lang/cargo/pull/13793, which only warns when each file handle points to a different file. The fix is scheduled on 1.79.0. You could try with the current beta toolchain.

**Comment by weihanglo** [comment]
Closing as resolved via #13793. Let us know if it turns out not.
